### PR TITLE
Miscellaneous Small Updates

### PR DESCRIPTION
**Pull request by krishmaj**
- after editing an entity type form, users are now redirected back to the page listing all entity types forms
- trashed component types are now only visible to users with permission to edit type forms
- improved formatting of the lists of roles and permissions on the user profile and permissions error pages
- small changes to the tooltip and error message text when using the Component UUID Formio box
- made the collapsible panel's title bold font (was previously normal font)
- small changes to text for various buttons and links on the page for editing entity type forms

### CODE REVIEW

File: app/pug/user_permissionError.pug [hunk 1]
```diff
@@ -5,37 +5,33 @@ head
...
-            |  (User ID: 
             .monospace #{required_user.user_id}
-            | )
-      
+
       p Your currently assigned roles are: 
         br
-        .monospace #{JSON.stringify(user.roles)}
-      
+        .monospace #{JSON.stringify(user.roles).replace(/"/g, "").replace(/,/g, ", ").replace(/^\[|]/g, '')}
```
Comment (micchickenburger) [inline app/pug/user_permissionError.pug]: You could slightly simplify:  `.replace(/["\[\]]/g, '').replace(/,/g, ', ');`

File: app/pug/user_permissionError.pug [hunk 2]
```diff
@@ -5,37 +5,33 @@ head
...
+
       p These roles give you the following permissions: 
         br
-        .monospace #{JSON.stringify(user.permissions).replace(/,/g, ", ")}
-      
+        .monospace #{JSON.stringify(user.permissions).replace(/"/g, "").replace(/,/g, ", ").replace(/^\[|]/g, '')}
```
Comment (micchickenburger) [inline app/pug/user_permissionError.pug]: You could slightly simplify:  `.replace(/["\[\]]/g, '').replace(/,/g, ', ');`

File: app/pug/user_profile.pug [hunk 3]
```diff
@@ -44,10 +44,10 @@ block content
         
         dt Roles
         dd
-          .monospace #{JSON.stringify(user.roles)}
+          .monospace #{JSON.stringify(user.roles).replace(/"/g, "").replace(/,/g, ", ").replace(/^\[|]/g, '')}
```
Comment (micchickenburger) [inline app/pug/user_profile.pug]: Seeing as these regexes are used across files, might be more maintainable to create a helper for this purpose

File: app/static/pages/edit_form.js [hunk 4]
```diff
@@ -2,6 +2,17 @@ var builder   = null;
...
+if(collection === 'componentForms')
+{
+  entityRoute = 'components/types';
+}
+
+if(collection === 'actionForms')
+{
+  entityRoute = 'actiontypes/list';
+}
+
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js:8]: This block can be simplified:

```javascript
let entityRoute;
if (collection === 'componentForms') entityRoute = 'components/types';
if (collection === 'actionForms') entityRoute = 'actiontypes/list';
```

File: app/static/pages/edit_form.js [hunk 5]
```diff
@@ -240,9 +251,17 @@ function SubmitData(submission)
     contentType: 'application/json',
     url:  "/json/" + collection + "/" + formId,
     data: JSON.stringify(submission.data),
-    success: schemaRecordChange
+    success: postSuccess
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js]: Prefer comma at end of line

File: app/static/pages/edit_form.js [hunk 6]
```diff
@@ -240,9 +251,17 @@ function SubmitData(submission)
...
   })
-   .fail(function(res, statusCode, statusMsg)
+   .fail(postFail);
+  
+  function postSuccess(result)
+  {
+    schemaRecordChange;
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js]: This statement has no effect

File: app/static/pages/edit_form.js [hunk 7]
```diff
@@ -240,9 +251,17 @@ function SubmitData(submission)
...
+    window.location.href = '/' + entityRoute;
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js]: Prefer backtick notation

File: app/static/pages/edit_form.js [hunk 8]
```diff
@@ -240,9 +251,17 @@ function SubmitData(submission)
...
     url:  "/json/" + collection + "/" + formId,
     data: JSON.stringify(submission.data),
-    success: schemaRecordChange
+    success: postSuccess
   })
-   .fail(function(res, statusCode, statusMsg)
+   .fail(postFail);
+  
+  function postSuccess(result)
+  {
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js:252]: Function braces should be on same line as function declaration

File: app/static/pages/edit_form.js [hunk 9]
```diff
@@ -240,9 +251,17 @@ function SubmitData(submission)
     contentType: 'application/json',
     url:  "/json/" + collection + "/" + formId,
     data: JSON.stringify(submission.data),
-    success: schemaRecordChange
+    success: postSuccess
   })
-   .fail(function(res, statusCode, statusMsg)
+   .fail(postFail);
+  
+  function postSuccess(result)
```
Comment (micchickenburger) [inline app/static/pages/edit_form.js]: `result` is unused in function

**Review by micchickenburger — DISMISSED** [review]
There might be a bug in the code.  Please review and use your discretion.  Approving so you're not waiting on me.

File: app/pug/user_profile.pug (hunk 3 above)
Comment (krishmaj) [reply app/pug/user_profile.pug]: Added a new pug mixin to cover all of the above `.replace` simplifications.

File: app/static/pages/edit_form.js (hunk 4 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js:8]: Changed.

File: app/static/pages/edit_form.js (hunk 5 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js]: Changed.

File: app/static/pages/edit_form.js (hunk 6 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js]: Removed, and confirmed that everything still working correctly.

File: app/static/pages/edit_form.js (hunk 7 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js]: Changed.

File: app/static/pages/edit_form.js (hunk 8 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js:252]: This whole file needs a proper cleanup anyway, which I plan to do as part of a future priority task, so I've left all the brackets and other formatting for now.

File: app/static/pages/edit_form.js (hunk 9 above)
Comment (krishmaj) [reply app/static/pages/edit_form.js]: Removed.

**Pull request merged**